app/src/main/java/com/example/todolistproject/view/TaskActivity.kt

```diff
@@ -0,0 +1,243 @@
+        val description = dialogViewBinding.taskDescriptionEditText.text.toString()
+        val startDate = dialogViewBinding.startDateEditText.text.toString()
+        val endDate = dialogViewBinding.endDateEditText.text.toString()
+
+        val task =
+            Task(
+                id = null,
+                isCompleted = false,
+                title = title,
+                description = description,
+                startDate = startDate,
+                endDate = endDate
+            )
+        mViewModel.insertTask(task)
+    }
+
+    private fun openModifyTaskDialog(task: Task) {
+        val dialogViewBinding = DialogAddTaskBinding.inflate(LayoutInflater.from(this))
+        dialogViewBinding.task = task
+
+        val dialog = AlertDialog.Builder(this).setTitle(getString(R.string.todo_edit_task))
+            .setView(dialogViewBinding.root)
+            .setPositiveButton(getString(R.string.todo_okay)) { dialog, _ ->
+                modifyTask(dialogViewBinding, task)
+                dialog.dismiss()
+            }
+            .setNegativeButton(getString(R.string.todo_cancel)) { dialog, _ ->
+                dialog.dismiss()
+            }
+            .create()
+        dialog.show()
+    }
+
+    private fun modifyTask(dialogViewBinding: DialogAddTaskBinding, task: Task) {
+        // We must deep copy the task object in order for the DiffUtil to recognize and compare the
+        // new object with the old one.
+        val modifiedTask = task.copy()
+        modifiedTask.title = dialogViewBinding.taskTitleEditText.text.toString()
+        modifiedTask.description = dialogViewBinding.taskDescriptionEditText.text.toString()
+        modifiedTask.startDate = dialogViewBinding.startDateEditText.text.toString()
+        modifiedTask.endDate = dialogViewBinding.endDateEditText.text.toString()
+
+        mViewModel.updateTask(modifiedTask)
+    }
+
+    private fun openDeleteTaskDialog(task: Task) {
+        val dialog = AlertDialog.Builder(this).setTitle(getString(R.string.todo_delete_task))
+            .setMessage(getString(R.string.todo_delete_task_confirmation))
+            .setPositiveButton(getString(R.string.todo_yes)) { _, _ ->
+                mViewModel.deleteTask(task)
+            }.setNegativeButton(getString(R.string.todo_no), null).create()
+        dialog.show()
+    }
+
+    private fun setTaskCompleted(task: Task) {
+        val isCompleted = task.isCompleted == true
+        task.isCompleted = !isCompleted
+
+        mViewModel.updateTask(task)
+    }
+
+    private fun openDatePickerDialog(editText: TextInputEditText) {
+        val calendar = Calendar.getInstance()
+        val dateSetListener = DatePickerDialog.OnDateSetListener { _, year, month, day ->
+            val date = Calendar.getInstance()
+            date.set(Calendar.YEAR, year)
+            date.set(Calendar.MONTH, month)
+            date.set(Calendar.DAY_OF_YEAR, day)
+            setDateText(editText, date)
```

Comment:
```suggestion
            editText.setText(date.convertBy("yyyy-MM-dd"))
```
    private fun Calendar.convertBy(format:String): CharSequence {
        val sdf = SimpleDateFormat(format, Locale.getDefault())
        return sdf.format(time)
    }

p2; 이런방법이 더 명확해 보여요
function에 인자 A, B를 넣고 그 안에서 A에 set 하는 방식은 좋은 방식이 아닙니다.
아마 그건 이펙티브 자바에 나와있었던거로 기억해요.

Reply:
좋은 팁 감사합니다. 말씀하신 대로 수정해서 bugfix/review_comments 브랜치에 적용 시켰습니다!